src/scenarios/personalization/ArrayAddFavorites/ArrayAddFavorites.tsx

```diff
@@ -0,0 +1,245 @@
+        id: id,
+        title: title,
+      };
+
+      const result = await apolloClient.mutate({
+        mutation: setArrayItemMutation,
+        variables: {
+          input: {
+            scope: 'PROFILE',
+            key: 'user_profile:favorites',
+            value: entityObject,
+          },
+        },
+        context: {
+          headers: {
+            Authorization: `Bearer ${userAccessToken}`,
+          },
+        },
+        fetchPolicy: 'no-cache',
+      });
+
+      if (result.errors) {
+        logger.error(
+          `method [${addToFavorites.name}]`,
+          'output:',
+          result.errors,
+        );
+      } else {
+        logger.log(`method [${addToFavorites.name}]`, 'output:', result.data);
+      }
+    } catch (error) {
+      if (error instanceof Error) {
+        logger.error(
+          `method [${addToFavorites.name}]`,
+          'output:',
+          error.message,
+        );
+      } else {
+        logger.error(
+          `method [${addToFavorites.name}]`,
+          'output:',
+          JSON.stringify(error),
+        );
+      }
+    }
+  };
+
+  return (
+    <Segment basic>
+      <Header size="huge">Array: Add Favorite(s)</Header>
+      <Header size="small">
+        Required Services:
+        <Label>user-service</Label>
+        <Label>catalog-service</Label>
+        <Label>personalization-service</Label>
+      </Header>
+
+      <Divider />
+
+      <Container fluid>
+        <p>
+          The scenario gives an example on how to mark favorite items of the
+          user.
+        </p>
+        <p>
+          In a real-life scenario any kind of list of items that is linked to a
+          user&apos;s profile can be managed (eg. favorites, watch history,
+          to-do list, etc)
+        </p>
+        <p>
+          If the user is not already signed-in, you can use one of the Sign-In
+          scenarios.
+        </p>
+      </Container>
+
+      <Divider />
+
+      <Grid divided>
+        <Grid.Column width={8}>
+          <Segment basic>
+            <Form>
+              <Form.Input
+                control={VariableSearch}
+                icon="key"
+                label="User Access Token"
+                value={userAccessToken}
+                setStateValue={setUserAccessToken}
+              />
+
+              <Form.Button
+                primary
+                onClick={async () => {
+                  fetchAllCatalogItems();
+                }}
+              >
+                Fetch All Catalog Items
+              </Form.Button>
+
+              <Divider />
+
+              <Form.Dropdown
+                fluid
+                selection
+                label="Entity ID"
+                placeholder="Select an Entity Type: Title"
+                options={entityArray.map((entity) => {
+                  return {
+                    text: `${entity.type}: ${entity.title}`,
+                    value: `${entity.id}: ${entity.title}`,
```

Comment:
I would suggest to have only the `entity.id` as the value here.

We need to change `entityId` state as follows:

from:
`const [entityId, setSelectedEntity] = useState<string>('');`
to:
`const [selectedEntity, setSelectedEntity] = useState<Entity>();`

Then in `onChange` event of the dropdown, you will get the id of the entity as the value. use the id to find the entity from `entityArray` and set it using `setSelectedEntity`

Then in `addToFavorites` method, we already know the id and the title of the selected entity (which is stored in `selectedEntity`). Use it to call the mutation.